### PR TITLE
add example data

### PR DESCRIPTION
**Pull request by hoppanda**

### CODE REVIEW

**Review by MikeJSeo — APPROVED** [review]

File: data-raw/dummy_anchored.R [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+library(flexsurv)
+set.seed(2024)
+
+# create adsl_twt
+adsl <- read.csv(system.file("extdata", "adsl.csv",
+  package = "maicplus",
+  mustWork = TRUE
+))
+adsl$USUBJID <- paste0("xx", adsl$USUBJID)
+adsl2 <- adsl
```
Comment (gravesti) [inline data-raw/dummy_anchored.R:13]: Do we need to set the ARM here?
```suggestion
adsl2 <- adsl
adsl2$ARM <- "C"
```

Comment (MikeJSeo) [reply data-raw/dummy_anchored.R:13]: Well.. adtte has arm name "A". Code probably wants to simulate control arm "C"

File: R/data_unanchored.R [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+#' @source These datasets are generated by the dummy_unanchored.R script.
+#'
+#' @keywords dataset
+#'
+"adsl_sat"
+"adtte_sat"
+"pseudo_ipd_sat"
+"agd_sat"
+"centered_ipd_sat"
+"adrs_sat"
```
Comment (gravesti) [inline R/data_unanchored.R:20]: Normally we'd need to document each data set, about all the columns and what they contain.

File: R/data.R [hunk 3]
```diff
@@ -0,0 +1,89 @@
+#' Dummy Single-arm Trial Data for Unanchored MAIC
+#'
+#' These datasets contains adsl_sat, adtte_sat, adrs_sat,
+#' agd_sat (AgD of effect modifiers), pseudo_ipd_sat (AgD, tte data)
+#' @format **adsl_sat**: a data frame with xx rows and x columns:
+#'   \describe{
+#'     \item{id}{The ID of the patients.}
+#'   }
```
Comment (gravesti) [inline R/data.R:8]: We can have all the anchored data sets documented in one page. What do you think? Same for unanchored
@hoppanda

File: data-raw/dummy_anchored.R [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+pseudo_ipd_twt <- rbind(pseudo_ipd, pseudo_ipd2)
+
+# create centered adsl_twt
+agd <- process_agd(agd)
+adsl_twt <- dummize_ipd(adsl_twt, dummize_cols = c("SEX"), dummize_ref_level = c("Female"))
+centered_ipd_twt <- center_ipd(ipd = adsl_twt, agd = agd)
+
+
+### Output
+usethis::use_data(adsl_twt, adtte_twt, pseudo_ipd_twt, centered_ipd_twt,
```
Comment (gravesti) [inline data-raw/dummy_anchored.R:62]: Need adrs for anchored example?

File: R/data.R [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+#'
+#' @format **pseudo_ipd_twt**: a data frame with xx rows and x columns:
+#'   \describe{
+#'     \item{id}{The ID of the patients.}
+#'   }
+#' @format **centered_ipd_twt**: a data frame with xx rows and x columns:
+#'   \describe{
+#'     \item{id}{The ID of the patients.}
+#'   }
+#'
```
Comment (gravesti) [inline R/data.R:76]: add agd in both anchored and unanchored

File: data-raw/dummy_unanchored.R [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+))
+# for time-to-event endpoints, pseudo IPD from digitalized KM
+pseudo_ipd_sat <- read.csv(system.file("extdata", "psuedo_IPD.csv",
+  package = "maicplus",
+  mustWork = TRUE
+))
+pseudo_ipd_sat$ARM <- "B"
+
+### Centered IPD
+agd <- process_agd(agd)
```
[the remark this answers is not in the document]
Comment (MikeJSeo) [reply data-raw/dummy_unanchored.R]: If we still have the extdata/aggregate_data_example_1.csv file, then we can probably just save the processed one. In the example agd and process_agd(agd) are the same btw.

File: R/data.R [hunk 7]
```diff
@@ -0,0 +1,205 @@
...
+#'     \item{USUBJID}{Unique subject identifiers for patients.}
+#'     \item{ARM}{Assigned treatment arm.}
+#'     \item{AGE}{Age in years at baseline.}
+#'     \item{SEX}{Sex of patient recorded as character `"Male"`/`"Female"`.}
+#'     \item{SMOKE}{Smoking status at baseline as integer `0`/`1`.}
+#'     \item{ECOG0}{ECOG score at baseline as integer `0`/`1`.}
+#'     \item{N_PR_THER}{Number of prior therapies received as integer `1, 2, 3, 4`.}
+#'     \item{SEX_MALE}{Indicator of `SEX == "Male"` as numeric `1`/`0`.}
+#'     \item{AGE_CENTERED}{Age in years at baseline relative to average in aggregate data [agd].}
+#'     \item{AGE_MEDIAN_CENTERED}{Age greater/less than median age in [agd] coded as  TODO .}
```
Comment (MikeJSeo) [inline R/data.R:63]: Age exceeding the comparator population median age are assigned a value of 1, while values lower than
the median age are assigned 0. Then this binary variable is centered by 0.5.

File: R/data.R [hunk 8]
```diff
@@ -0,0 +1,205 @@
...
+#'     \item{AGE_SQUARED_CENTERED}{TODO}
```
Comment (MikeJSeo) [inline R/data.R:64]: Age is squared and centered by the aggregate data E(X^2) which is found through variance formula

File: R/data.R [hunk 9]
```diff
@@ -0,0 +1,205 @@
...
+#'     \item{SEX_MALE_CENTERED}{TODO}
```
Comment (MikeJSeo) [inline R/data.R:65]: Sex of patient centered by the aggregate data proportion of male patients

File: R/data.R [hunk 10]
```diff
@@ -0,0 +1,205 @@
...
+
+#' Centered patient data from single arm trial
+#' @format A data frame with 500 rows and 14 columns:
+#'   \describe{
+#'     \item{USUBJID}{Unique subject identifiers for patients.}
+#'     \item{ARM}{Assigned treatment arm.}
+#'     \item{AGE}{Age in years at baseline.}
+#'     \item{SEX}{Sex of patient recorded as character `"Male"`/`"Female"`.}
+#'     \item{SMOKE}{Smoking status at baseline as integer `0`/`1`.}
+#'     \item{ECOG0}{ECOG score at baseline as integer `0`/`1`.}
```
Comment (MikeJSeo) [inline R/data.R:59]: Indicator of `ECOG score == 0` at baseline as numeric `1`/`0`

File: R/data.R [hunk 11]
```diff
@@ -0,0 +1,205 @@
...
+
+#' Patient data from single arm study
+#' @format a data frame with 500 rows and 8 columns:
+#'   \describe{
+#'     \item{USUBJID}{Unique subject identifiers for patients.}
+#'     \item{ARM}{Assigned treatment arm.}
+#'     \item{AGE}{Age in years at baseline.}
+#'     \item{SEX}{Sex of patient recorded as character `"Male"`/`"Female"`.}
+#'     \item{SMOKE}{Smoking status at baseline as integer `0`/`1`.}
+#'     \item{ECOG0}{ECOG score at baseline as integer `0`/`1`.}
```
Comment (MikeJSeo) [inline R/data.R:11]: Indicator of ECOG score == 0 at baseline as numeric 1/0

File: R/data.R [hunk 12]
```diff
@@ -0,0 +1,205 @@
...
+#'     \item{USUBJID}{Unique subject identifiers for patients.}
+#'     \item{ARM}{Assigned treatment arm, `"A"`.}
+#'     \item{EVNT}{Event indicator `0`/`1`.}
+#'     \item{AVAL}{Analysis value which in this dataset overall survival time in days.}
+#'     \item{AVALU}{Unit of `AVAL`.}
+#'     \item{PARAMCD}{Paramater code of `AVAL`, `"OS"`.}
+#'     \item{PARAM}{Parameter name of `AVAL`, `"Overall Survival`.}
+#'     \item{CNSR}{Censoring indicator `0`/`1`.}
+#'     \item{TIME}{Survival time in days.}
+#'     \item{EVENT}{Event indicator `0`/`1`.}
```
Comment (MikeJSeo) [inline R/data.R:32]: EVENT and EVNT are same is that okay? I guess so?

File: R/data.R [hunk 13]
```diff
@@ -0,0 +1,205 @@
...
+#'     \item{SEX}{Sex of patient recorded as character `"Male"`/`"Female"`.}
+#'     \item{SMOKE}{Smoking status at baseline as integer `0`/`1`.}
+#'     \item{ECOG0}{ECOG score at baseline as integer `0`/`1`.}
+#'     \item{N_PR_THER}{Number of prior therapies received as integer `1, 2, 3, 4`.}
+#'     \item{SEX_MALE}{Indicator of `SEX == "Male"` as numeric `1`/`0`.}
+#'     \item{AGE_CENTERED}{Age in years at baseline relative to average in aggregate data [agd].}
+#'     \item{AGE_MEDIAN_CENTERED}{Age greater/less than median age in [agd] coded as  TODO .}
+#'     \item{AGE_SQUARED_CENTERED}{TODO}
+#'     \item{SEX_MALE_CENTERED}{TODO}
+#'     \item{ECOG0_CENTERED}{TODO}
```
Comment (MikeJSeo) [inline R/data.R:66]: ECOG0 centered by the aggregate data proportion of ECOG0

File: R/data.R [hunk 14]
```diff
@@ -0,0 +1,205 @@
...
+#'     \item{SMOKE_CENTERED}{TODO}
```
Comment (MikeJSeo) [inline R/data.R:67]: Smoke status centered by the aggregate data proportion of smokers

File: R/data.R [hunk 15]
```diff
@@ -0,0 +1,205 @@
...
+#' This data is formatted to be used in [center_ipd()].
+#'
+#' @format A data frame with 3 rows and 9 columns:
+#'   \describe{
+#'     \item{STUDY}{The study name, Study_XXXX}
+#'     \item{ARM}{Study arm name or total}
+#'     \item{N}{Number of observations in study arm}
+#'     \item{AGE_MEAN}{Mean age in study arm}
+#'     \item{AGE_MEDIAN}{Median age in study arm}
+#'     \item{AGE_SD}{TODO}
```
Comment (MikeJSeo) [inline R/data.R:113]: Median standard deviation of age in study arm?

File: R/data.R [hunk 16]
```diff
@@ -0,0 +1,205 @@
...
+#'     \item{SEX_MALE_COUNT}{Number of male patients}
+#'     \item{ECOG0_COUNT}{Number of patients with ECOG score TODO}
```
Comment (MikeJSeo) [inline R/data.R:115]: Number of patients with ECOG score of 0

File: R/data.R [hunk 17]
```diff
@@ -0,0 +1,205 @@
...
+
+#' Centered patient data from two arm trial
+#' @format A data frame with 1000 rows and 14 columns:
+#'   \describe{
+#'     \item{USUBJID}{Unique subject identifiers for patients.}
+#'     \item{ARM}{Assigned treatment arm.}
+#'     \item{AGE}{Age in years at baseline.}
+#'     \item{SEX}{Sex of patient recorded as character `"Male"`/`"Female"`.}
+#'     \item{SMOKE}{Smoking status at baseline as integer `0`/`1`.}
+#'     \item{ECOG0}{ECOG score at baseline as integer `0`/`1`.}
```
Comment (MikeJSeo) [inline R/data.R:182]: ECOG0 centered by the aggregate data proportion of ECOG0

File: R/data.R [hunk 18]
```diff
@@ -0,0 +1,205 @@
...
+#'     \item{N_PR_THER}{Number of prior therapies received as integer `1, 2, 3, 4`.}
+#'     \item{SEX_MALE}{Indicator of `SEX == "Male"` as numeric `1`/`0`.}
+#'     \item{AGE_CENTERED}{Age in years at baseline relative to average in aggregate data [agd].}
+#'     \item{AGE_MEDIAN_CENTERED}{Age greater/less than median age in [agd] coded as  TODO .}
```
Comment (MikeJSeo) [inline R/data.R:186]: same as above

File: R/data.R (hunk 12 above)
Comment (gravesti) [reply R/data.R:32]: Remove `EVNT` and check code works

File: data-raw/dummy_anchored.R (hunk 4 above)
Comment (gravesti) [reply data-raw/dummy_anchored.R:65]: @gravesti simulate some data here

**Pull request merged**